sway/desktop/render.c

```diff
@@ -242,22 +242,26 @@ struct fx_framebuffer *get_main_buffer_blur(struct fx_renderer *renderer, struct
 
 	int blur_radius = config->blur_params.radius;
 	int blur_passes = config->blur_params.num_passes;
+	float blur_noise = config->blur_params.noise;
 
 	// Downscale
 	for (int i = 0; i < blur_passes; ++i) {
 		wlr_region_scale(&tempDamage, &damage, 1.0f / (1 << (i + 1)));
 		render_blur_segments(renderer, gl_matrix, &tempDamage, &current_buffer,
-				&renderer->shaders.blur1, box, blur_radius);
+				&renderer->shaders.blur1, box, blur_radius, blur_noise);
 	}
 
 	// Upscale
 	for (int i = blur_passes - 1; i >= 0; --i) {
 		// when upsampling we make the region twice as big
 		wlr_region_scale(&tempDamage, &damage, 1.0f / (1 << i));
 		render_blur_segments(renderer, gl_matrix, &tempDamage, &current_buffer,
-				&renderer->shaders.blur2, box, blur_radius);
+				&renderer->shaders.blur2, box, blur_radius, blur_noise);
 	}
 
+	render_blur_segments(renderer, gl_matrix, &tempDamage, &current_buffer,
```

Comment:
I'd pass `damage` instead of `&tempDamage`. I couldn't see any damage artifacts but this might solve it on your end?

Reply:
I'll give this a try, thanks

Reply:
Didn't solve the damage artifacting, but I did make the change and result seems to be the same